src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRun/LocalMethodImport.cs

```diff
@@ -51,7 +51,10 @@ public override IEnumerable<DependencyListEntry> GetStaticDependencies(NodeFacto
             {
                 yield return entry;
             }
-            yield return new DependencyListEntry(_localMethod, "Local method import");
+            if (!Method.IsAggressiveOptimization)
```

Comment:
I think this is still going to generate the method body into the image, but not put it into the table. I'll defer to Tomas, but it feels like we should use the same mechanism that is used to skip uncompilable methods so that the body is not generated at all (MethodWithGCInfo returning true from IsEmpty).